File: Algorithms/JavaScript/minimum-depth-of-binary-tree.js
/**
 * Given a binary tree, find its minimum depth.
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 *
 * @param {TreeNode} treeNode
 * @return {number}
 */

var minimumDepthOfBinaryTree = function(treeNode) {
  return treeNode === null ?
    0 :
    1 + Math.min(minimumDepthOfBinaryTree(treeNode.left), minimumDepthOfBinaryTree(treeNode.right));
};

File: Algorithms/Ruby/minimum_depth_of_binary_tree.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} treeNode
# @return {Integer}

def minimum_depth_of_binary_tree(tree_node)
  tree_node.nil? ?
    0 :
    1 + [minimum_depth_of_binary_tree(tree_node.left), minimum_depth_of_binary_tree(tree_node.right)].min
end
